tests/unit/test_base_adapter.py

```diff
@@ -10,94 +10,93 @@
-             schema=".*",
-             name="(?i)^.*(?<!_view)$"),
-        dict(database="snowshu_development",
-             schema="source_system",
-             name="order_items_view")
-    ]
-
-    mock_filtered_schema = [
-        BaseSQLAdapter._DatabaseObject("SOURCE_SYSTEM", Relation("snowshu_development", "source_system", "", None, None)),
-        BaseSQLAdapter._DatabaseObject("Cased_Schema", Relation("snowshu_development", "Cased_Schema", "", None, None)),
-    ]
-
-    included_relations = [
-        # included tables
-        Relation("snowshu_development", "source_system", "fake_table_1", mz.TABLE, []),
-        Relation("snowshu_development", "Cased_Schema", "fake_table_2", mz.TABLE, []),
-        # included view
-        Relation("snowshu_development", "source_system", "order_items_view", mz.VIEW, []),
-    ]
-    excluded_relations = [
-        # excluded _view
-        Relation("snowshu_development", "source_system", "some_other_view", mz.VIEW, []),
-        Relation("snowshu_development", "Cased_Schema", "another_view", mz.VIEW, []),
-    ]
-
+    stubbed_adapter.credentials = creds
+
+    assert stubbed_adapter._build_conn_string() == \
+        f'postgres://{creds.user}:{creds.password}@{creds.host}/{creds.database}?account={creds.account}'
+
+@pytest.mark.parametrize("config_patterns, mock_filtered_schema, included_relations, excluded_relations", [
+    (
+        [
+            dict(database="snowshu_development", schema=".*", name="(?i)^.*(?<!_view)$"),
+            dict(database="snowshu_development", schema="source_system", name="order_items_view")
+        ],
+        [
+            BaseSQLAdapter._DatabaseObject("SOURCE_SYSTEM", Relation("snowshu_development", "source_system", "", None, None)),
+            BaseSQLAdapter._DatabaseObject("Cased_Schema", Relation("snowshu_development", "Cased_Schema", "", None, None)),
+        ],
+        [
+            Relation("snowshu_development", "source_system", "fake_table_1", mz.TABLE, []),
+            Relation("snowshu_development", "Cased_Schema", "fake_table_2", mz.TABLE, []),
+            Relation("snowshu_development", "source_system", "order_items_view", mz.VIEW, []),
+        ],
+        [
+            Relation("snowshu_development", "source_system", "some_other_view", mz.VIEW, []),
+            Relation("snowshu_development", "Cased_Schema", "another_view", mz.VIEW, []),
+        ]
+    )
+])
+def test_build_catalog(stubbed_adapter, config_patterns, mock_filtered_schema, included_relations, excluded_relations):
     mock_relations = included_relations + excluded_relations
-    def mock_get_relations_func(schema_obj: BaseSQLAdapter._DatabaseObject):
+    def mock_get_relations_func(schema_obj: StubbedAdapter._DatabaseObject):
         return [r for r in mock_relations if schema_obj.full_relation.schema == r.schema]
 
-    # stubbed version of the BaseSourceAdapter with the required class vars
-    class StubbedSourceAdapter(BaseSourceAdapter):
```

Comment:
@eculicny do you know what's the exact reason of having this class here? I refactored class a little, seems to be working as expected but for now removed it as didn't see a point, but maybe I'm missing something?

Reply:
Currently, I'm not sure what the original reason was. Probably either setup for future tests or tests that no longer exist.

I haven't checked, but I assume it isn't referenced anywhere else.

Reply:
No, this is the only place it exists. I see no reference to that